src/AsmResolver/Shims/MemoryMappedFileShim.Windows.cs

```diff
@@ -0,0 +1,74 @@
+using System;
+using System.ComponentModel;
+using System.Runtime.InteropServices;
+using System.Threading;
+
+namespace AsmResolver.Shims;
+
+internal sealed unsafe partial class MemoryMappedFileShim
+{
+    private const uint GENERIC_READ = 0x80000000;
+    private const uint FILE_SHARE_READ = 0x2;
+    private const uint OPEN_EXISTING = 3;
+    private const uint PAGE_READONLY = 2;
+    // not on msdn, stole from terrafx
+    private const uint FILE_MAP_READ = 4;
+
+    private void* _fileHandle;
+    private void* _mappingHandle;
+
+    [DllImport("kernel32", ExactSpelling = true, CharSet = CharSet.Unicode, SetLastError = true)]
+    private static extern nint CreateFileW(char* name, uint access, uint shareMode, void* securityAttributes, uint create, uint attributes);
+
+    [DllImport("kernel32", ExactSpelling = true, SetLastError = true)]
+    [return: MarshalAs(UnmanagedType.Bool)]
+    private static extern bool GetFileSizeEx(void* handle, out long size);
+
+    [DllImport("kernel32", ExactSpelling = true, SetLastError = true)]
+    [return: MarshalAs(UnmanagedType.Bool)]
+    private static extern bool CloseHandle(void* handle);
+
+    [DllImport("kernel32", ExactSpelling = true, CharSet = CharSet.Ansi, SetLastError = true)]
+    private static extern nint CreateFileMappingA(void* handle, void* mapAttributes, uint protection, uint maxSizeHigh,
+        uint maxSizeLow, sbyte* name);
+
+    [DllImport("kernel32", ExactSpelling = true, SetLastError = true)]
+    private static extern void* MapViewOfFile(void* handle, uint access, uint offsetHigh, uint offsetLow, nuint size);
```

Comment:
The typing is inconsistent here.  The `handle` parameters are of type `void*`, while the APIs responsible for creating these handles return `nint`. Can't we just do `nint` for all handles (as is typically the convention)?

Reply:
My logic was pretty much
 - if the invalid/error return is -1, use nint so it can be compared to that
- else if the invalid return is null, just use void*
if you'd like to me to standardize on one or the other, I don't mind either